### ISSUE TITLE
Work out weaklaplacian(::DirichletPolynomial) for non-evenly spaced points

### ISSUE THREAD

**Issue by dlfivefifty** [JuliaApproximation/PiecewiseOrthogonalPolynomials.jl#39]
I think the only thing that's not trivial is the `A` block. The `D` blocks should just be changing `step(r)` to `diff(r)`

**Comment by dlfivefifty** [comment]
Note the `A` block is just standard FEM with hat functions. @ioannisPApapadopoulos do you know this off the top of your head?

**Comment by ioannisPApapadopoulos** [comment]
Yeah should be quite simply, something like -1/r for the off-diagonal and 1/r0 + 1/r1 for the main diagonal, where r0 and r1 are the interval lengths on the left and right hand side of the hats.

E.g. if a hat was scaled onto -1/2..0, 0..5, then the off diagonals are

(2 * -2)/2 = -2. = -1/r0 (left)
(-1/5 * 1/5)* 5 = -1/5 = =1/r1 (right).

the main diagonal is
(2*2)/2 + (-1/5 * -1/5)*5 = 2 + 1/5 = 1/r0 + 1/r1 = 11/5.

**Comment by dlfivefifty** [comment]
Ah ok so it’ll be something like ` 1 ./ diff(r)[1:end-1] + 1 ./ diff(r)[2:end] ` ? I think the dimensions match at least
